sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/StreamingJoinHelper.scala

```diff
@@ -256,12 +256,12 @@ object StreamingJoinHelper extends PredicateHelper with Logging {
           val castedLit = lit.dataType match {
             case CalendarIntervalType =>
               val calendarInterval = lit.value.asInstanceOf[CalendarInterval]
-              if (calendarInterval.months > 0) {
+              if (calendarInterval.months > 0 || calendarInterval.days > 0) {
```

Comment:
This can be breaking change. Why did you reject interval with days? By the way, `days` can be converted to microseconds

Reply:
done, I restore the constraint back